x-pack/plugins/security_solution/server/endpoint/migrations/turn_off_policy_protections.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger, ElasticsearchClient } from '@kbn/core/server';
+import type { UpdatePackagePolicy } from '@kbn/fleet-plugin/common';
+import type { AuthenticatedUser } from '@kbn/security-plugin/common';
+import {
+  isPolicySetToEventCollectionOnly,
+  setPolicyToEventCollectionOnly,
+} from '../../../common/endpoint/models/policy_config_helpers';
+import type { PolicyData } from '../../../common/endpoint/types';
+import { AppFeatureSecurityKey } from '../../../common/types/app_features';
+import type { EndpointInternalFleetServicesInterface } from '../services/fleet';
+import type { AppFeatures } from '../../lib/app_features';
+import { getPolicyDataForUpdate } from '../../../common/endpoint/service/policy';
+
+export const turnOffPolicyProtections = async (
+  esClient: ElasticsearchClient,
+  fleetServices: EndpointInternalFleetServicesInterface,
+  appFeaturesService: AppFeatures,
+  logger: Logger
+): Promise<void> => {
+  const log = logger.get('endpoint', 'policyProtections');
+
+  if (appFeaturesService.isEnabled(AppFeatureSecurityKey.endpointPolicyProtections)) {
+    log.info(
+      `App feature [${AppFeatureSecurityKey.endpointPolicyProtections}] is enabled. Nothing to do!`
+    );
+
+    return;
+  }
+
+  log.info(
+    `App feature [${AppFeatureSecurityKey.endpointPolicyProtections}] is disabled. Checking endpoint integration policies for compliance`
+  );
+
+  const { packagePolicy, internalSoClient, endpointPolicyKuery } = fleetServices;
+  const updates: UpdatePackagePolicy[] = [];
+  const messages: string[] = [];
+  let hasMoreData = true;
+  let total = 0;
+  let page = 1;
+
+  do {
+    const { items, total: totalPolicies } = await packagePolicy.list(internalSoClient, {
+      page: page++,
+      kuery: endpointPolicyKuery,
+      perPage: 1000,
+    });
+
+    total = totalPolicies;
+    hasMoreData = items.length > 0;
```

Comment:
this causes one additional call for `packagePolicy.list()` returning zero items. to avoid this, `total` could be used

update: I see you even have mocks prepared in the tests for the additional call. is it maybe intended?

Reply:
Its just another way to ensure all policies are covered case policies are added as this function is running. I think I might have picked up this approach from code I saw in fleet.

Reply:
sorry for nitpicking, but a small optimisation for keeping the startup time a bit lower is still an optimisation. I still don't think the extra fetch is needed, because if a new policy added at this point, it will go through the checks that ensure that no protections are enabled.

(what I don't know: is it even possible for policies to be added during `Plugin.start()`?)

in case it is possible and the checks are not in place when adding: I think the extra fetch still wouldn't help, only maybe for every N-thousand-and-first-some policies, while still possibly skipping the previous N-thousand minus-some policies (e.g. if the 3992...4016th policies were added after fetching the 4th page, 3992...4000th policies are skipped)

///

independently of the extra fetch, `sortOrder: 'asc'` and `sortField: 'createdAt'` should be added to the options for `packagePolicy.list()` as the default is descending by `updated_at`.

Reply:
Good point. I'll make the change to ensure we don't make the additional API call

👍